main/acle.md

```diff
@@ -7499,6 +7653,1415 @@ This is because the ACLE intrinsic calls do not imply a particular
+    entry to the function. Either the function does not change the ZA
+    state at all, or the function undoes any changes to the ZA state
+    before returning.
+
+    Note: C and C++ calls from shared-ZA functions to private-ZA
+    functions are defined to preserve ZA state. Such calls do not affect
+    whether a function can be marked [`arm_preserves_za`](#arm_preserves_za).
+
+*   If the function is [private-ZA](#private-za), the attribute guarantees
+    that a call to the function does not directly or indirectly involve
+    any of the following:
+
+    *   a call to an [`arm_new_za`](#arm_new_za) function
+
+    *   a call to `setjmp`
+
+    *   a call to [`__arm_za_disable`](#__arm_za_disable)
+
+    *   a call to a non-C/C++ function that “commits a lazy save”, in the
+        sense of the [[APCS64]](#AAPCS64)
+
+    *   an inline asm that commits a lazy save
+
+    *   the catching of an exception
+
+    The platform may place additional requirements as well.
+
+In both cases, the onus is on the definition of the function to honor
+the guarantee that is being made. The attribute does not direct the
+compiler to do anything to honor the guarantee.
+
+If a function with an `arm_preserves_za` type does not preserve ZA,
+the behavior is undefined. (There is an analogy with functions
+that are declared `noreturn` but do in fact return, and to functions
+that are declared `const` but do in fact change memory.)
+
+The attribute is intended to be useful for functions at API boundaries,
+where the compiler might not have access to the definition of the
+function being called. As the description above implies, attaching
+`arm_preserved_za` to a [private-ZA](#private-za) function is quite
+a low-level feature, but it is useful for
+[streaming-compatible versions of standard routines](#streaming-compatible-versions-of-standard-routines)
+and could be useful for things like vector math routines.
+
+Function types with this attribute implicitly convert to function types
+that do not have the attribute. The reverse is not true. For example:
+
+``` c
+  #define ATTR __attribute__((arm_preserves_za))
+
+  ATTR void (*ptr1)(void);
+  void (*ptr2)(void);
+  ATTR void f1(void);
+  void f2(void);
+
+  void code() {
+    ptr1 = ptr2;  // Ill-formed
+    ptr1 = f1;    // OK
+    ptr1 = f2;    // Ill-formed
+
+    ptr2 = ptr1;  // OK
+    ptr2 = f1;    // OK
+    ptr2 = f2;    // OK
+  }
+```
+
+## SME functions and intrinsics
+
+[`<arm_sme.h>`](#arm_sme.h) declares various support functions and
+defines various intrinsics. The support functions have
+[external linkage](#external-linkage), like standard C functions such
+as `memcpy` do. However, as noted in [Intrinsics](#intrinsics), it is
+unspecified whether the intrinsics are functions and, if so, what
+linkage they have.
+
+There are many more intrinsics than support functions, so everything
+described in this section is an intrinsic unless otherwise noted.
```

Comment:
```suggestion
There are many more intrinsics than support functions, so unless otherwise noted, everything
described in the following section is an intrinsic .
```

Reply:
I'd prefer to keep “this section” rather than “the following section”, since the intrinsic and function documentation is in (multiple) subsections of the current one.

I've changed the “unless otherwise noted” to “unless its individual description says otherwise” to try to make it less woolly.